sway-core/src/ir_generation/function.rs

```diff
@@ -1461,23 +1461,25 @@ impl<'te> FnCompiler<'te> {
         // We're dancing around a bit here to make the blocks sit in the right order.  Ideally we
         // have the cond block, followed by the body block which may contain other blocks, and the
         // final block comes after any body block(s).
+        //
+        // NOTE: This is currently very important!  There is a limitation in the register allocator
```

Comment:
Damn it, is this hitting us again? Do we plan on enforcing SSA eventually or just looking for a register allocator algorithm that is less fussy?

Reply:
This is exactly the problem I mentioned to you a couple of weeks ago, about the allocator not using topological traversal for liveness analysis.  But before we fix that it would really make sense to translate/preserve the basic blocks from the IR to the abstract ASM.